horace_core/GUI/horace.m

```diff
@@ -1436,52 +1455,32 @@ function Cut_do_cut_pushbutton_Callback(hObject, eventdata, handles)
 else
     try
         %must strip out square brackets, if user has inserted them:
-        s1=strfind(a1,'['); s2=strfind(a1,']');
-        if isempty(s1) && isempty(s2)
-            a1new=textscan(a1,'%f','delimiter',',');
-        elseif ~isempty(s1) && ~isempty(s2)
-            a1=a1(s1+1:s2-1);
-            a1new=textscan(a1,'%f','delimiter',',');
-        else
+        [a1new,ok] = read_vector(a1);
+        if ~ok
             mess1='   Ensure binning values are entered if the form of lo,step,hi / step / lo,hi    ';
```

Comment:
Is this not the `report_error` function below's job?

Reply:
report error picks up the message and report it to user over GUI and command window. Here we form the message specific fro the inputs we are processing.